File: VB/WpfApplication1/MainWindowViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace WpfApplication1
    <POCOViewModel> _
    Public Class MainWindowViewModel
        Inherits ViewModelBase


        Private report_Renamed As XtraReport1
        Public ReadOnly Property Report() As XtraReport
            Get
                Return report_Renamed
            End Get
        End Property
        Public Property DataSet() As nwindDataSet
        Public ReadOnly Property Categories() As DataTable
            Get
                Return DataSet.Tables(0)
            End Get
        End Property
        Public Sub New()
            If IsInDesignMode Then
                Return
            End If
            report_Renamed = New XtraReport1() With {.RequestParameters = False}
            DataSet = New nwindDataSet()
            report_Renamed.categoriesTableAdapter.Fill(DataSet.Categories)
        End Sub
    End Class
End Namespace

File: CS/WpfApplication1/MainWindowViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WpfApplication1 {
    [POCOViewModel]
    public class MainWindowViewModel : ViewModelBase {
        XtraReport1 report;
        public XtraReport Report { get { return report; } }
        public nwindDataSet DataSet { get; set; }
        public DataTable Categories { get { return DataSet.Tables[0]; } }
        public MainWindowViewModel() {
            if(IsInDesignMode)
                return;
            report = new XtraReport1() { RequestParameters = false };
            DataSet = new nwindDataSet();
            report.categoriesTableAdapter.Fill(DataSet.Categories);
        }
    }
}
